aten/src/ATen/core/LegacyTypeDispatch.cpp

```diff
@@ -9,4 +9,19 @@ LegacyTypeDispatch & globalLegacyTypeDispatch() {
   return singleton;
 }
 
+AT_DEFINE_REGISTRY(LegacyTypeInitRegistry, LegacyTypeInitInterface, LegacyTypeInitArgs)
+
+const LegacyTypeInitInterface& getLegacyTypeInit() {
+  static std::unique_ptr<LegacyTypeInitInterface> legacy_type_init;
+  static std::once_flag once;
+  std::call_once(once, [] {
```

Comment:
why is this necessary?

Reply:
This is the "standard" boilerplate we've been using for the other hooks mechanisms (e.g., CUDAHooks and VariableHooks). The setting is this: you have been given a registry which, when invoked, constructs an object (whose constructor was specified out of band by some other source). So using this you can construct an object. But you only want to construct the object once, and then reuse it later! Thus `call_once`.